Tests/SubprocessTests/ProcessMonitoringTests.swift

```diff
@@ -0,0 +1,361 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+#if canImport(System)
+@preconcurrency import System
+#else
+@preconcurrency import SystemPackage
+#endif
+
+#if canImport(Darwin)
+import Darwin
+#elseif canImport(Glibc)
+import Glibc
+#elseif canImport(Bionic)
+import Bionic
+#elseif canImport(Musl)
+import Musl
+#elseif canImport(WinSDK)
+import WinSDK
+#endif
+
+import Testing
+import Dispatch
+import Foundation
+import TestResources
+import _SubprocessCShims
+@testable import Subprocess
+
+@Suite("Subprocess Process Monitoring Unit Tests", .serialized)
+struct SubprocessProcessMonitoringTests {
+
+    init() {
+        #if os(Linux)
+        _setupMonitorSignalHandler()
+        #endif
+    }
+
+    private func immediateExitProcess(withExitCode code: Int) -> Configuration {
+        #if os(Windows)
+        return Configuration(
+            executable: .name("cmd.exe"),
+            arguments: ["/c", "exit \(code)"]
+        )
+        #else
+        return Configuration(
+            executable: .path("/bin/sh"),
+            arguments: ["-c", "exit \(code)"]
+        )
+        #endif
+    }
+
+    private func longRunningProcess(withTimeOutSeconds timeout: Double? = nil) -> Configuration {
+        #if os(Windows)
+        let waitTime = timeout ?? 99999
+        return Configuration(
+            executable: .name("powershell.exe"),
+            arguments: ["-Command", "Start-Sleep -Seconds \(waitTime)"]
+        )
+        #else
+        let waitTime = timeout != nil ? "\(timeout!)" : "infinite"
```

Comment:
to avoid the force unwrap:

```suggestion
        let waitTime = timeout.map { "\($0)" } ?? "infinite"
```